Tests/AWSSDKSwiftTests/Services/SQS/SQSTests.swift

```diff
@@ -0,0 +1,134 @@
+                try client.deleteQueue(request).wait()
+            }
+        }
+    }
+
+    //MARK: TESTS
+    
+    func testSendReceiveAndDelete() {
+        attempt {
+            let testData = try TestData(#function, client: client)
+
+            let messageBody = "Testing, testing,1,2,1,2"
+            let sendMessageRequest = SQS.SendMessageRequest(messageBody: messageBody, queueUrl: testData.queueUrl)
+            let messageId = try client.sendMessage(sendMessageRequest).wait().messageId
+            
+            // receive message tests the flattened arrays in XML response
+            var foundMessage = false
+            var messages : [SQS.Message] = []
+            repeat {
+                let receiveMessageRequest = SQS.ReceiveMessageRequest(maxNumberOfMessages: 10, queueUrl: testData.queueUrl)
+                messages = try client.receiveMessage(receiveMessageRequest).wait().messages ?? []
+                for message in messages {
+                    if messageId == message.messageId {
+                        if let body = message.body {
+                            foundMessage = true
+                            XCTAssertEqual(body, messageBody)
+                        }
+                        if let receiptHandle = message.receiptHandle {
+                            let deleteRequest = SQS.DeleteMessageRequest(queueUrl: testData.queueUrl, receiptHandle: receiptHandle)
+                            try client.deleteMessage(deleteRequest).wait()
+                        }
+                    }
+                }
+            } while messages.count > 0
+            XCTAssertTrue(foundMessage)
+        }
+    }
+    
+    func testGetQueueAttributes() {
+        attempt {
+            let testData = try TestData(#function, client: client)
+
+            let request = SQS.GetQueueAttributesRequest(attributeNames:[.queuearn], queueUrl: testData.queueUrl)
+            let result = try client.getQueueAttributes(request).wait()
+            XCTAssertNotNil(result.attributes?[.queuearn])
+        }
+    }
+    
+    func testTestPercentEncodedCharacters() {
+        attempt {
+            let testData = try TestData(#function, client: client)
+
+            let messageBody = "!@#$%^&*()-=_+[]{};':\",.<>\\|`~éñà"
+            let sendMessageRequest = SQS.SendMessageRequest(messageBody: messageBody, queueUrl: testData.queueUrl)
+            let messageId = try client.sendMessage(sendMessageRequest).wait().messageId
+            
+            // receive message tests the flattened arrays in XML response
+            var foundMessage = false
+            var messages : [SQS.Message] = []
+            repeat {
+                let receiveMessageRequest = SQS.ReceiveMessageRequest(maxNumberOfMessages: 10, queueUrl: testData.queueUrl)
+                messages = try client.receiveMessage(receiveMessageRequest).wait().messages ?? []
+                for message in messages {
```

Comment:
```swift
XCTAssertEqual(messages.count, 1)
guard let message = messages.first else { XCTFail("expected to get the first message") }

XCTAssertEqual(message.body, messageBody)

if let receiptHandle = message.receiptHandle {
  let deleteRequest = SQS.DeleteMessageRequest(queueUrl: testData.queueUrl, receiptHandle: receiptHandle)
  try client.deleteMessage(deleteRequest).wait()
}
else {
  XCTFail("expected to have a receipt handle")
}
```

Reply:
slight difference, I used XCTUnwrap

Reply:
Didn't know about `XCTUnwrap` that looks great!